### PR TITLE
Register user

### PR DESCRIPTION
**Pull request by aparker22**
Registers users by sending email and password.  Receives token, but we can change it to sending user to login screen if needed.

@NickWilsonDev @illiaChaban

### CODE REVIEW

File: Components/RegisterScreen.js [hunk 1]
```diff
@@ -1,77 +1,112 @@
...
+        this.state = { email: '', password: '' };
+      }
 
-let RegisterScreen = ({props}) =>
-<View style={styles.container}>
+    render() {
+        let {navigation, dispatch} = this.props;
+        let {email, password} = this.state;
+
+        let registerUser = () => {
```
Comment (illiaChaban) [inline Components/RegisterScreen.js:19]: i dont really to understand this part.

File: Components/RegisterScreen.js [hunk 2]
```diff
@@ -1,77 +1,112 @@
...
+            submitNewUserInformation(email, password)
+            .then(res => dispatch(updateUserObject(res)))
+            .then(dispatch(updateIsUserLoggedIn()))
+            .then(this.setState({email: '', password: ''}))
+            .then(navigation.navigate('Home'))
```
Comment (illiaChaban) [inline Components/RegisterScreen.js:24]: why would you need to save email and password in state

File: Components/RegisterScreen.js (hunk 1 above)
Comment (illiaChaban) [reply Components/RegisterScreen.js:19]: Also it won't register the user, because back-end requires username too

Comment (aparker22) [reply Components/RegisterScreen.js:19]: Ok, I thought that we were using email for the username.

Which part do you not understand?  Local state (the component state) is only saving the value of the input fields and then resets it once the submit button is pressed.  The redux store (this.props) is saving the navigation props and the dispatch actions.

File: Components/RegisterScreen.js (hunk 2 above)
Comment (aparker22) [reply Components/RegisterScreen.js:24]: The local component is saving the state of the textinput components.  It resets it to an empty string once the submit button is pressed.

Comment (illiaChaban) [reply Components/RegisterScreen.js:24]: lol, I read setState, but i didn't get to empty strings. I guess GraphQL weared me down

**Pull request merged**